src/ray/util/sequencer.h

```diff
@@ -0,0 +1,49 @@
+#ifndef RAY_UTIL_SEQUENCER_H_
+#define RAY_UTIL_SEQUENCER_H_
+
+#include <deque>
+#include <unordered_map>
+#include "absl/synchronization/mutex.h"
+
+namespace ray {
+
+/// \class Sequencer
+/// Sequencer guarantees that all operations with the same ordering key are sequenced.
+/// This class is thread safe.
+template <class KEY>
+class Sequencer {
+ public:
+  void execute_ordered(KEY key, std::function<void()> operation) {
```

Comment:
1) document public methods.
2) should we CamelCase for class methods.

Reply:
fixed